### PR TITLE
Rename disabled file on mod duplication

### LINKED ISSUE #550: Disabling a mod and redownloading it creates duplicate entries stuck as enabled and disabled
### Operating System

Linux

### Description of bug

I tried to update a mod that is currently conflicting with my other mods. It's disabled and there wasn't an update for it so I got the same mod downloaded, but PolyMC began showing both the enabled and disabled versions on the mod list.  

There's no way to disable this duplicate enabled entry, and of course deleting it brings back the old behavior. 

### Steps to reproduce

1. Download a mod that you know breaks your MC
2. Disable it
3. Try redownloading it because you're not sure if there's a new update
4. The mod of the same version is downloaded
5. Now two versions of the same mod exist, an enabled mod and a disabled mod
6. The disabled mod cannot be enabled and the enabled mod cannot be disabled 

### Suspected cause

PolyMC renames mods to mark them as disabled, so when a new version of the same mod is downloaded it prevents the disabled mod from renaming itself back to normal and prevents the enabled mod from renaming itself to disabled.

I think PolyMC should keep track of mods separately and create and remove symlinks to have better control on how disabling currently works.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
fixes https://github.com/PolyMC/PolyMC/issues/550
To test:
- download a mod 
- disable it (it will appear disabled)
- download the same mod again, the same version
- no more file duplication because of name (not in the folder and not in the list) as we delete one of the resources

Drawbacks: replaces the old file

### CODE REVIEW

**Comment by ColonelGerdauf** [comment]
Question: When a "new" mod is installed, and the existing mod is disabled, is the new mod also disabled?

**Comment by Trial97** [comment]
No, as you chose to download it again.

File: launcher/minecraft/mod/tasks/BasicFolderLoadTask.h [hunk 1]
```diff
@@ -49,10 +49,16 @@ class BasicFolderLoadTask : public Task {
...
             auto resource = m_create_func(entry);
-            resource->moveToThread(m_thread_to_spawn_into);
-            m_result->resources.insert(resource->internal_id(), resource);
+            if (names.contains(resource->name())) {
+                resource->destroy();
+            } else {
+                names << resource->name();
+                resource->moveToThread(m_thread_to_spawn_into);
+                m_result->resources.insert(resource->internal_id(), resource);
+            }
```
Comment (Ryex) [inline launcher/minecraft/mod/tasks/BasicFolderLoadTask.h]: as simple and foolproof as this seems I'm not sure this is either the correct place for this logic or even the correct logic.

as is, two resources with the same name (even if they are different versions or even different resources) cause conflicts, but silently deleting the second resource of the same name found seems wrong. renaming it to something like `<name>.duplicateN` and highlight it in the list seems better.

in the case of PolyMC/#550 and the downloading a duplicate mod. is seems like better logic to explicitly look for the expected `<name>.disabled` file during the download and handling that case in the download logic.

**Review by Ryex — CHANGES_REQUESTED** [review]

File: launcher/minecraft/mod/tasks/BasicFolderLoadTask.h (hunk 1 above)
Comment (Trial97) [reply launcher/minecraft/mod/tasks/BasicFolderLoadTask.h]: Just to not forget: moving the logic to download mod will not suffice.
You can still add the same mod from external sources(and also the Add Mod button).
And this also affects all the other resources so I would love to implement it as low as possible to not have to implement it for each view.
Other than that your suggestion with duplicate suffix I will make it.

Comment (Trial97) [reply launcher/minecraft/mod/tasks/BasicFolderLoadTask.h]: Now it renames the files instead of removing them

**Review by Ryex — APPROVED** [review]

**Review by timoreo22 — APPROVED** [review]

**Pull request merged**